### PR TITLE
Update for MAUI Preview 11

### PR DESCRIPTION
**Pull request by brminnick**
## Description

Updates `CommunityToolkit.Maui.Markup` to support .NET MAUI Preview 11

### General
- Update `.editorconfig` to match `CommunityToolkit.Maui`
- Update XML Documentation
  - Add `CS1574` `CS1573` `CS137` to `<WarningsAsErrors>` in `Directory.Build.props`
- Add `dotnet-format-daily.yml` workflow
  - Daily GitHub Action to use `dotnet format` and open a formatting Pull Request
  - Copied from [`CommunityToolkit.Maui`](https://github.com/CommunityToolkit/Maui/blob/main/.github/workflows/dotnet-format-daily.yml)

### CommunityToolkit.Maui.Markup
- Remove Support for `LayoutOptions.*AndExpand`
  - .NET MAUI has Deprecated the the following `LayoutOptions`:
     - `LayoutOptions.StartAndExpand`
     - `LayoutOptions.CenterAndExpand`
     - `LayoutOptions.EndAndExpand`
- Use `ArgumentNullException.ThrowIfNull` when applicable
- Change `== null` to `is null`

### CommunityToolkit.Maui.Markup.UnitTests
- Remove Unit Tests for `LayoutOptions.*AndExpand`
- Remove `mdoc` NuGet Package
  - This was a relic from XCT and we're not currently using it in `CommunityToolkit.Maui`
- Update NuGet Packages
- Use `ArgumentNullException.ThrowIfNull` when applicable
- Remove `public` modifier from classes
- Rename `MarkupTestFixtureBase` to `BaseMarkupTestFixture`
- Change `== null` to `is null`

### CommunityToolkit.Maui.Markup.Sample
- Change `Color.FromHex` to `Color.FromArgb`
- Update NuGet Packages

### CODE REVIEW

File: samples/CommunityToolkit.Maui.Markup.Sample/Pages/Base/BasePage.cs [hunk 1]
```diff
@@ -5,11 +5,11 @@ namespace CommunityToolkit.Maui.Markup.Sample.Pages.Base;
 
 abstract class BaseContentPage<T> : ContentPage where T : BaseViewModel
```
Comment (VladislavAntonyuk) [inline samples/CommunityToolkit.Maui.Markup.Sample/Pages/Base/BasePage.cs:6]: Let's rename to BasePage or rename file to BaseContentPage

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseMarkupTestFixture.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using Microsoft.Maui.Controls;
+using NUnit.Framework;
+
+namespace CommunityToolkit.Maui.Markup.UnitTests.Base;
+
+class BaseMarkupTestFixture<TBindable> : BaseMarkupTestFixture where TBindable : BindableObject, new()
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseMarkupTestFixture.cs:7]: We can make it abstract

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Globalization;
+using CommunityToolkit.Maui.Markup.UnitTests.Mocks;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Dispatching;
+using NUnit.Framework;
+
+namespace CommunityToolkit.Maui.Markup.UnitTests.Base;
+
+public class BaseTestFixture
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs:10]: We can make it abstract

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Globalization;
+using CommunityToolkit.Maui.Markup.UnitTests.Mocks;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Dispatching;
+using NUnit.Framework;
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs:6]: Maybe we can rewrite tests using xUnit. But it is not necessary in this PR

File: src/CommunityToolkit.Maui.Markup.UnitTests/BindableLayoutExtensionsTests.cs [hunk 5]
```diff
@@ -1,82 +1,83 @@
 ﻿using System;
+using CommunityToolkit.Maui.Markup.UnitTests.Base;
 using Microsoft.Maui.Controls;
 using NUnit.Framework;
 
 namespace CommunityToolkit.Maui.Markup.UnitTests;
 
 [TestFixture]
-public class BindableLayoutExtensionsTests : MarkupBaseTestFixture<StackLayout>
+class BindableLayoutExtensionsTests : BaseMarkupTestFixture<StackLayout>
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui.Markup.UnitTests/BindableLayoutExtensionsTests.cs:9]: Test class should be public

**Review by VladislavAntonyuk — APPROVED** [review]

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseMarkupTestFixture.cs (hunk 2 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseMarkupTestFixture.cs:7]: Done ✅

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs (hunk 3 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs:10]: done ✅

File: src/CommunityToolkit.Maui.Markup.UnitTests/BindableLayoutExtensionsTests.cs (hunk 5 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.UnitTests/BindableLayoutExtensionsTests.cs:9]: In NUnit, test classes don't need to be public.

I Xunit, test classes **do** need to be public.

File: src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs (hunk 4 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.UnitTests/Base/BaseTestFixture.cs:6]: I'm not against it.

We re-write the `CommunityToolkit.Maui` Unit Tests in Xunit to match .NET MAUI's Unit Tests.

Xunit may also make it easier to write [Device Tests](https://github.com/dotnet/maui/tree/main/src/Core/tests/DeviceTests).

File: samples/CommunityToolkit.Maui.Markup.Sample/Pages/Base/BasePage.cs (hunk 1 above)
Comment (brminnick) [reply samples/CommunityToolkit.Maui.Markup.Sample/Pages/Base/BasePage.cs:6]: Done ✅

**Pull request merged**